chef/automate
# Adding error messages to the search bar

![Screen Shot 2019-05-23 at 4 13 11 PM](https://user-images.githubusercontent.com/1679247/58292430-2c0a5b80-7d76-11e9-9565-3c6e4b6460a7.png)

![Screen Shot 2019-05-23 at 4 17 26 PM](https://user-images.githubusercontent.com/1679247/58292451-45aba300-7d76-11e9-85ec-4b57bc295451.png)


### :nut_and_bolt: Description

Allowing error messages to be filter on in the Client Runs's search bar. This adds a new field in the node-state called 'error_message' this field contains the error.message field in the original CCR. The best way to use this feature is with regex. If a user has the error of 'file[/failed/file/resource] (insights-test::default line 26) had an error: Chef::Exceptions::EnclosingDirectoryDoesNotExist: Parent directory /failed/file does not exist.', they can search for just a portion of the message. For example search with `*Chef::Exceptions::EnclosingDirectoryDoesNotExist*` or `*/failed/file does not exist*`. 

### :athletic_shoe: Demo Script / Repro Steps

1. `build components/config-mgmt-service && build components/ingest-service && build components/automate-ui`
1. `start_all_services`
1. `send_chef_run_failure_example`
1. Open https://a2-dev.test/client-runs
1. In the search bar select type 'Error'
1. Type 'does not exist' and ensure a suggestion appears
1. Then create a filter for `*Chef::Exceptions::EnclosingDirectoryDoesNotExist*` and ensure that one node is displayed in the node list. 
1. Remove all filters and create a new Error filter with `*/failed/file does not exist*` and ensure that one node is displayed in the node list. 

### :chains: Related Resources

https://github.com/chef/automate/issues/368

### :white_check_mark: Checklist
